app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/UserPhotosViewAdapterTest.kt

```diff
@@ -1,32 +1,110 @@
 
+import android.graphics.Bitmap
+import android.view.View
 import android.widget.LinearLayout
 import androidx.fragment.app.commit
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.testing.TestLifecycleOwner
 import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
 import androidx.test.platform.app.InstrumentationRegistry
 import com.github.sdp_begreen.begreen.R
 import com.github.sdp_begreen.begreen.activities.MainActivity
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
 import com.github.sdp_begreen.begreen.models.ParcelableDate
+import com.github.sdp_begreen.begreen.models.ProfilePhotoMetadata
 import com.github.sdp_begreen.begreen.models.TrashCategory
 import com.github.sdp_begreen.begreen.models.TrashPhotoMetadata
+import com.github.sdp_begreen.begreen.models.User
 import com.github.sdp_begreen.begreen.rules.KoinTestRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.test.runTest
 import org.hamcrest.CoreMatchers
 import org.hamcrest.CoreMatchers.equalTo
 import org.hamcrest.MatcherAssert
+import org.junit.BeforeClass
 import org.junit.Rule
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.mockito.Mockito
+import org.mockito.kotlin.whenever
 
+@RunWith(AndroidJUnit4::class)
+@LargeTest
 class UserPhotosViewAdapterTest {
+
+    companion object {
+        private val db: DB = Mockito.mock(DB::class.java)
+        private val auth: Auth = Mockito.mock(Auth::class.java)
+        private val userId = "10"
+        private val profilePhotoMetadata = ProfilePhotoMetadata("1", null, userId)
+        val user = User(userId, 2, "test", 5, "test",
+            "test", "test", 15, listOf("1", "3", "6"), listOf("2", "4"), profilePhotoMetadata)
+        val users = listOf(
+            User("1", 123, "Alice"),
+            User("2", 0, "Bob Zeu bricoleur"),
+            User("3", 14, "Charlie Chaplin"),
+            User("4", 23, "David Pujadas"),
+            User("5", 10492, "Euler"),
+            User("6", 1234, "Alain Berset"),
+            User("7", 1235, "Mister Alix")
+        )
+        private val fakePicture = Bitmap.createBitmap(120, 120, Bitmap.Config.ARGB_8888)
+        private val authUserFlow = MutableStateFlow(userId)
+
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        @BeforeClass
+        @JvmStatic
+        fun setUp() {
+            // The implementation need to be provided before the rule is executed,
+            // that's why we do it in the beforeClass method
+            runTest {
+                // setup basic get user and getProfilePicture use in multiple tests
+                Mockito.`when`(db.getUser(user.id))
```

Comment:
As I said previouly, I don't undersand what you mean by import it statically since I think I did the same as other test classes. Could you enlighten me a bit please ?

Reply:
I made the change